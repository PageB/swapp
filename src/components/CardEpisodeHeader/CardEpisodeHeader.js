import React from 'react';
import PropTypes from 'prop-types';

import styles from './CardEpisodeHeader.module.scss';

const cardEpisodeHeader = props => {
  return (
    <div className={[styles.Card, styles[props.theme]].join(' ')}>
      <img className={styles.Card__Image} src={props.card.image} alt={props.card.title} />
      <div className={styles.Card__Details}>
        <p className={styles.Card__Title}>{props.card.episode}</p>
        <p className={styles.Card__SubTitle}>{props.card.title}</p>
      </div>
    </div>
  );
};

cardEpisodeHeader.propTypes = {
  theme: PropTypes.oneOf(['DarkTheme', 'LightTheme']),
};

cardEpisodeHeader.defaultProps = {
  theme: 'DarkTheme',
};

export default cardEpisodeHeader;
