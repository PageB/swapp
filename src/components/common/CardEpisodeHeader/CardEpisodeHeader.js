import React from 'react';

import styles from './CardEpisodeHeader.module.scss';

const cardEpisodeHeader = props => {
  return (
    <div className={[styles.Card, styles[props.theme]].join(' ')}>
      <img className={styles.CardImage} src={props.card.image} alt={props.card.title} />
      <div className={styles.CardDetails}>
        <p className={styles.CardEpiside}>{props.card.episode}</p>
        <p className={styles.CardTitle}>{props.card.title}</p>
      </div>
    </div>
  );
};

export default cardEpisodeHeader;
