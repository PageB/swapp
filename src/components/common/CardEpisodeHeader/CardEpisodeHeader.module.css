.Card {
  background-color: #333333;
  border-radius: 8px;
  color: #ffe300;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  margin: 16px;
}

.CardImage {
  border-radius: 8px 0 0 8px;
  width: 260px;
}

.CardDetails {
  color: inherit;
  background-color: inherit;
  padding: 0 16px;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.CardEpiside {
  color: inherit;
  font-weight: bold;
  font-size: 36px;
  padding: 0;
  margin: 0;
}

.CardTitle {
  border-radius: 8px 8px 0px 0px;
  color: inherit;
  font-weight: bold;
  font-size: 20px;
  color: #4bd5ee;
}

.DarkTheme {
  background-color: #333333;
  color: #ffe300;
}

.DarkTheme .CardTitle {
  color: #4bd5ee;
}

.LightTheme {
  background-color: #ffffff;
  color: #4bd5ee;
}

.LightTheme .CardTitle {
  color: #333333;
}
