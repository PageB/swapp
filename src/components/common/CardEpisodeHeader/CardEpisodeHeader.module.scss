@import '../../../colors.scss';
@import '../../../constants.scss';

.DarkTheme {
  background-color: $dark-charcoal;
  color: $vivid-yellow;

  .Card__Title {
    color: $crayola;
  }
}

.LightTheme {
  background-color: $white;
  color: $crayola;

  .Card__Title {
    color: $dark-charcoal;
  }
}

.Card {
  border-radius: 8px;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin: 16px;

  &__Image {
    border-radius: 8px 0 0 8px;
    max-width: $card-width;
    flex: 1;
  }

  &__Details {
    color: inherit;
    background-color: inherit;
    padding: 0 16px;
    border-radius: 8px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    flex: 6;
  }

  &__Title {
    color: inherit;
    font-weight: bold;
    font-size: 36px;
    padding: 0;
    margin: 0;
  }

  &__Subtitle {
    border-radius: 8px 8px 0px 0px;
    color: inherit;
    font-weight: bold;
    font-size: 20px;
    color: $crayola;
  }
}
