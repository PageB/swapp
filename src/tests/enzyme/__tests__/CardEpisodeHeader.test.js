import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { starWarsEpisodes } from '../../../mock/data';

import CardEpisodeHeader from '../../../components/CardEpisodeHeader/CardEpisodeHeader';

configure({ adapter: new Adapter() });

describe('CardEpisodeHeader Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CardEpisodeHeader card={starWarsEpisodes[0]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
