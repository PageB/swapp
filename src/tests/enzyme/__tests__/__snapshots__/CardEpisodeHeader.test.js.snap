// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardEpisodeHeader Component should match snapshot 1`] = `
<div
  className="Card DarkTheme"
>
  <img
    className="Card__Image"
  />
  <div
    className="Card__Details"
  >
    <p
      className="Card__Title"
    />
    <p
      className="Card__SubTitle"
    />
  </div>
</div>
`;
